File: Zaawansowane progrmowanie obiektowe, C++/lab 1/src/student.cpp
#include <iostream>
#include "person.h"

using namespace std;
class Student : public Person {
private:
    string index;
public:
    Student() {
        cout << "Konstruktor bez. klasy pochodnej – Student" << endl;
    }

    Student(string name1, string surname1, int age1, string index1) : Person(name1, surname1, age1) {
        index = index1;
        cout << "Konstruktor klasy pochodnej - Student" << endl;
    }

    void setIndex(string newIndex) {
        index = newIndex;
    }

    string getIndex() {
        return index;
    }

    void showInfoStudent() {
        cout << "Imie: " << name << " nazwisko: " << surname << "wiek: " << age << " index: " << index << endl;
    }
};

File: Podstawowe aplikacji w języku Java/Lab2/var/src/Student.java
public class Student {
    String name;
    int age;
    double mark;

    Student(){
        name = "Unknown Student";
        age = 0;
        mark = 0.0;
    }

    Student(String name, int age, double mark){
        this.name = name;
        this.age = age;
        this.mark = mark;
    }
}
